src/components/ImportJobs/ImportJobs.js

```diff
@@ -53,23 +54,34 @@ class ImportJobs extends Component {
   };
 
   onDrop = async acceptedFiles => {
+    const {
+      setFiles,
+      updateUploadDefinition,
+    } = this.context;
+
     this.setState({ isDropZoneActive: false });
 
-    const { setFiles } = this.context;
+    await updateUploadDefinition();
+
+    const { uploadDefinition } = this.context;
```

Comment:
Add this in the assignment on top.

Reply:
This line must be after `await updateUploadDefinition()` as it fetches upladDefinition and writes it in context.

Reply:
Got it. Let's make `updateUploadDefinition` to return the definition so it is not needed to read for it from context. Also, it will make easier dealing with errors more clearly.